plugins/handler/plugin_api_handler.go

```diff
@@ -14,9 +14,24 @@
 
 package handler
 
+const (
+	NetworkServiceResource         = "networkservice"
```

Comment:
I think relying on "const" for the type of object is not best approach, I think using golang type assertion is way more reliable.  In this case passing a single object to corresponding interface function would be sufficient.

Reply:
Are you suggesting I create an interface to pass this back, and have different types implement it and we can then use golang type assertion?

Reply:
No, when you call ObjectedCreated/ObjectedDelete just pass an original object, api method will do type assertion and call right private func.

Reply:
I'll push a new version addressing the remaining comments and then we can work through this one, I'm not fully understanding what you mean here.

Reply:
just an idea,
```
const (
	Add = iota
	Delete
	Update
)
type message struct{
 operation int
 obj interface{}
}
```
so we go back to shared informer, when any handler is called add/delete/update, you push on the queue `message` with operation and the original object. In the queue you call API method if operation `add`, you call ObjectedCreated with `message.obj` if operation `delete` then you call ObjectedDeleted with again `message.obj`. As you can see the code becomes object type independent and very simple.